vault12factor/__init__.py

```diff
@@ -226,10 +226,17 @@ def __init__(self, vaulturl: str, vaultauth: VaultAuthentication, secretpath: st
         self.pin_cacert = pin_cacert
         self.ssl_verify = ssl_verify
         self.debug_output = debug_output
-        self._cache = None  # type: Dict[str, str]
+        self._cache = {}  # type: Dict[str, str]
         self._leasetime = None  # type: datetime.datetime
         self._updatetime = None  # type: datetime.datetime
         self._lease_id = None  # type: str
+        self._refresh()
+
+    def _attach_secret_attribute(instance, attribute):
+        class_name = instance.__class__.__name__ + 'Child'
+        child_class = type(class_name, (instance.__class__,), {attribute: property(lambda self: self._get_or_update(attribute))})
```

Comment:
instead either shadow `self` here or rename this instance of the parameter name? This way this should be able to stay PEP-8 compliant, right?